Pass  'clip=False' argument to `random_noise` subcalls

jdarena66 opened the pull request:
## Description

<!--
- Reference relevant issues or related pull requests with their URL / #<number>.
- Use `pre-commit` to check and format code.
-->
Closes https://github.com/scikit-image/scikit-image/issues/7923

## Release note

For maintainers and optionally contributors, please refer to the [instructions](https://scikit-image.org/docs/stable/development/contribute.html#documenting-changes) on how to document this PR for the release notes.

```release-note
In `skimage.util.random_noise`, ensure that `clip` argument is handled
consistently for various modes.
```

stefanv commented on src/skimage/util/noise.py:
```diff
@@ -70,10 +70,10 @@ def random_noise(image, mode='gaussian', rng=None, clip=True, **kwargs):
-        for modes `'speckle'`, `'poisson'`, and `'gaussian'`. This is
-        needed to maintain the proper image data range. If False, clipping
-        is not applied, and the output may extend beyond the range [-1, 1].
+        If True (default), the output will be clipped after noise applied.
```
```suggestion
        If True (default), the output will be clipped after noise is applied.
```

stefanv reviewed (commented):
LGTM, thanks!

We'll need a test to verify the correct behavior.

The previous example snippet provided wasn't indicative of the problem, because the salt-pepper ratio was too low for any noise to be added. It still exhibited the clipping behavior, just without any noise being added (i.e., input image was being clipped).

jdarena66 commented:
@stefanv -- thanks for the info on RNGs! I rebased and added a test case called `test_clipping_application_consistency` within tests/skimage/util/test_random_noise.py, and now assert that setting `clip=False` doesn't perform upper or lower clipping with `mode` argument `salt` or `pepper`. 

Hope this looks good, but I'm also happy to make more updates if there are aspects of skimage testing practices that I haven't captured. 

Thanks!
